models/db.py

```diff
@@ -89,6 +89,26 @@
   migrate='runestone_courses.table'
 )
 
+
+# Provide a common query. Pass ``db.courses.ALL`` to retrieve all fields; otherwise, only the ``course_name`` and ``base_course`` are selected.
+def get_course_row(*args, **kwargs):
+    if not args:
+        args = db.courses.course_name, db.courses.base_course
+    return db(db.courses.id == auth.user.course_id).select(*args).first()
+
+
+# Provide the correct URL to a book, based on if it's statically or dynamically served. This function return URL(*args) and provides the correct controller/function based on the type of the current course (static vs dynamic).
+def get_course_url(*args):
+    # Redirect to old-style statically-served books if it exists; otherwise, use the dynamically-served controller.
+    if os.path.exists(os.path.join(request.folder, 'static', auth.user.course_name)):
+        return URL(c='static', args=[auth.user.course_name, *args])
```

Comment:
I'm getting syntax errors on this -- `args=[auth.user.course_name] + list(args)` does what I **think** you want...

Reply:
That sounds reasonable. I didn't test under Python 2, perhaps that's why I missed it.

Reply:
Yep, confirmed in CI -- 2.7 fails here, 3.7 passes it.